File: solutions/c/triangle/1/triangle.c
#include "triangle.h"

bool is_triangle(triangle_t sides) {
    if(sides.a >= 0 && sides.b >= 0 && sides.c >= 0) {
        if(sides.a + sides.b >= sides.c && sides.b + sides.c >= sides.a && sides.a + sides.c >= sides.b) {
            return true;
        }
    }
    return false;
}
bool is_equilateral(triangle_t sides) {
    if(is_triangle(sides) && sides.a == sides.b && sides.b == sides.c && sides.a > 0) {
        return true;
    }
    return false;
}
bool is_isosceles(triangle_t sides) {
    if(is_triangle(sides) && (sides.a == sides.b || sides.a == sides.c || sides.c == sides.b)) {
        return true;
    }
    return false;
}
bool is_scalene(triangle_t sides) {
    if(is_triangle(sides) && !is_isosceles(sides)) {
        return true;
    }
    return false;
}

File: solutions/julia/triangle/1/triangle.jl
triangle(a,b,c) = a > 0 && b > 0 && c > 0 && a+b >= c && a+c>=b && b+c >=a

function is_equilateral(sides)
    a = sides[1]
    b = sides[2]
    c = sides[3]
    return a > 0 && a == b && a == c
end

function is_isosceles(sides)
    a = sides[1]
    b = sides[2]
    c = sides[3]
    return triangle(a,b,c) && (a == b || b==c || a==c)
end

function is_scalene(sides)
    a = sides[1]
    b = sides[2]
    c = sides[3]
    return triangle(a,b,c) && !is_isosceles(sides)
end
